### PR TITLE
UIKit: If focused element hidden by keyboard, then change UIView bounds

### PR DESCRIPTION
**Pull request by dima-avdeev-jb**
On UIKit we have convenient method setBounds(...) and setClipsToBounds(true).
It hide's part of UIView and moves visible part to original coordinates with animation.

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt [hunk 1]
```diff
@@ -52,9 +122,16 @@ internal actual class ComposeWindow : UIViewController {
...
                 skikoUIView.hideScreenKeyboard()
             },
+            getTopLeftOffset = ::getTopLeftOffset,
         )
         layer.setContent(content = content)
-        layer.setSize(width.toInt(), height.toInt())
+    }
+
+    override fun viewWillAppear(animated: Boolean) {
+        println("traitCollection().displayScale: ${traitCollection().displayScale}")
```
Comment (igordmn) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt]: Let's don't leave logs in the jb-main code. We will definitely forget to remove them in the future :)

File: compose/ui/ui/src/jsNativeMain/kotlin/androidx/compose/ui/native/ComposeLayer.jsNative.kt [hunk 2]
```diff
@@ -129,6 +132,10 @@ internal class ComposeLayer(
         scene.constraints = Constraints(maxWidth = width, maxHeight = height)
     }
 
+    internal fun getActiveFocusRect():Rect? {
```
Comment (igordmn) [inline compose/ui/ui/src/jsNativeMain/kotlin/androidx/compose/ui/native/ComposeLayer.jsNative.kt]: nit: the class itself is internal, so we can omit internal here

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt [hunk 3]
```diff
@@ -32,15 +43,74 @@ internal actual class ComposeWindow : UIViewController {
...
+
+        @Suppress("unused")
+        @ObjCAction
+        fun keyboardDidHide(arg: NSNotification) {
+            view.setClipsToBounds(false)
+        }
+    }
+
+    init {
+        NSNotificationCenter.defaultCenter.addObserver(
```
Comment (igordmn) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt]: Shouldn't we call `removeObserver` in something like `dispose`? Or is it definitely weak-referenced, and we don't need to do that?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt]: done

File: compose/ui/ui/src/jsNativeMain/kotlin/androidx/compose/ui/native/ComposeLayer.jsNative.kt (hunk 2 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/jsNativeMain/kotlin/androidx/compose/ui/native/ComposeLayer.jsNative.kt]: done

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt (hunk 3 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt]: I will check

**Review by igordmn — APPROVED** [review]

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt (hunk 3 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/uikit/ComposeWindow.uikit.kt]: Good suggestion, I moved addObserver to viewWillAppear method. And removeObserver to viewDidDisappear method.
Now subscriptions cancels.

**Pull request merged**